src/pages/ProfilesExplorerView/infrastructure/labels/LabelsDataSource.ts

```diff
@@ -9,37 +9,30 @@ import {
 } from '@grafana/data';
 import { RuntimeDataSource, sceneGraph } from '@grafana/scenes';
 import { isPrivateLabel } from '@shared/components/QueryBuilder/domain/helpers/isPrivateLabel';
-import { labelsRepository } from '@shared/infrastructure/labels/labelsRepository';
+import { LabelsRepository } from '@shared/infrastructure/labels/labelsRepository';
+import { MemoryCacheClient } from '@shared/infrastructure/MemoryCacheClient';
 
 import { GroupByVariable } from '../../domain/variables/GroupByVariable/GroupByVariable';
 import { computeRoundedTimeRange } from '../../helpers/computeRoundedTimeRange';
 import { PYROSCOPE_LABELS_DATA_SOURCE } from '../pyroscope-data-sources';
 import { DataSourceProxyClientBuilder } from '../series/http/DataSourceProxyClientBuilder';
 import { LabelsApiClient } from './http/LabelsApiClient';
 
+// we instantiate a new repository here to prevent unwanted interaction with the QueryBuilder
+// indeed, when entering the "idle" state, the QueryBuilder state machine will call cancelAllLoad()
+// that will abort any ApiClient in-flight request
+const labelsRepository = new LabelsRepository({
+  apiClient: new LabelsApiClient({ dataSourceUid: '' }),
```

Comment:
Why is the uid forced to be empty here, compared to https://github.com/grafana/explore-profiles/blob/4ee541acbe822d92abfc9344eda4611600b1476e/src/shared/infrastructure/labels/labelsRepository.ts#L115-L118

Reply:
Because they are different class of API clients. This one, used by `LabelsDataSource` inherits from `DataSourceProxyClient`, which requires a data source UID when instantiated. This allows us to switch the data source in runtime when the user selects different data source in the UI.

The other one inherits from `ApiClient`, which determines the data source UID only when the user lands on the page for the 1st time. It's used by the (legacy) Comparison page where we don't need to switch the data source in runtime.

It's confusing, but it's temporal. Once we migrate the Comparison page, we'll only need the client inheriting from `DataSourceProxyClient`.

Reply:
Thanks for explaining, that makes sense 👍